### PR TITLE
Add change password and change nickname

### LINKED ISSUE #194: [User UI] Change Password
# Description
Provides a form for the user to change their password

### LINKED ISSUE #195: [User UI] Change Nickname
# Description
Provides a form to allow the user to change their nickname

### LINKED ISSUE #196: [User Service] Change Password Endpoint
# Description
Expose an endpoint on user service to support change password requests

### LINKED ISSUE #197: [User Service] Change Nickname Endpoint
# Description
Expose an endpoint on user-service to accept requests to change nickname

### PR DESCRIPTION
**Pull request by laughingkid-sg**
## Changes
- Add an endpoint on user service to support change password requests
- Add an endpoint on user service to accept requests to change the nickname
- Add respective UI for the above endpoints

## Related Issues
- Closes #197 
- Closes #196 
- Closes #195 
- Closes #194

### CODE REVIEW

File: frontend/src/components/form/ChangeNicknameForm.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      <Heading as="h5" size="sm">
+        Change Nickname
+      </Heading>
+      <form onSubmit={handleSubmit(validFormHandler, invalidFormHandler)}>
+        <FormControl id="nickname" isInvalid={!!errors.nickname}>
+          <FormLabel>New Nickname</FormLabel>
+
+          <Input
+            type="text"
+            {...register("nickname", {
```
Comment (KwanHW) [inline frontend/src/components/form/ChangeNicknameForm.tsx:95]: The validators can be moved to `constants/validator.ts` (see #203).

File: frontend/src/components/form/ChangePasswordForm.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+  return (
+    <VStack alignItems="flex-start">
+      <Heading as="h5" size="sm">
+        Change Password
+      </Heading>
+      <form onSubmit={handleSubmit(validFormHandler, invalidFormHandler)}>
+        <FormControl id="password" isInvalid={!!errors.password}>
+          <FormLabel>New Password</FormLabel>
+          <InputGroup>
```
Comment (KwanHW) [inline frontend/src/components/form/ChangePasswordForm.tsx:97]:
```suggestion
          <InputGroup>
          <PasswordInput register={register} formKey="password" />
```
1. This entire section has been abstracted to `<PasswordInput />` 
2. I think we should have a password confirmation too

File: frontend/src/components/ui/navbar/HomeNavbar.tsx [hunk 3]
```diff
@@ -41,37 +43,42 @@ function HomeNavbar() {
...
-                <MenuItem onClick={logoutHandler}>Logout</MenuItem>
-              </MenuList>
-            </Menu>
-          </HStack>
+              <Menu>
+                <MenuButton as={Button} variant="ghost" cursor="pointer">
+                  <Text fontSize="lg">{`Hello, ${user.nickname}!`}</Text>
+                </MenuButton>
+                <MenuList alignItems="center">
+                  <MenuItem onClick={() => onOpen()}>Account Settings</MenuItem>
```
Comment (KwanHW) [inline frontend/src/components/ui/navbar/HomeNavbar.tsx:72]:
```suggestion
                  <MenuItem onClick={() => onOpen()}>Account Settings</MenuItem>
                  <MenuItem onClick={onOpen}>Account Settings</MenuItem>
```
Very minor nit.

**Review by KwanHW — CHANGES_REQUESTED** [review]
Good effort on the frontend. Just need to update the changes from #203

**Review by sharpstorm — APPROVED** [review]
Backend LGTM

**Review by KwanHW — APPROVED** [review]
LGTM for frontend

**Pull request merged**